include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -919,6 +919,23 @@ def HasBeenResetIntrinsicOp : FIRRTLOp<"int.has_been_reset", [Pure]> {
   let assemblyFormat = "$clock `,` $reset attr-dict `:` type($reset)";
 }
 
+def FPGAProbeIntrinsicOp : FIRRTLOp<"int.fpga_probe", []> {
+  let summary = "Mark a value to be observed through FPGA debugging facilities";
+
+  let description = [{
+    The `firrtl.int.fpga_probe` intrinsic marks a value in the IR to be made observable
+    through FPGA debugging facilities. Most FPGAs offer a form of signal observation
+    or logic analyzer to debug a design. This operation allows the IR to indicate which
+    signals should be made observable for debugging. Later FPGA-specific passes
+    may then pick this information up and materialize the necessary logic analyzers
+    or tool scripts.
+  }];
+  let arguments = (ins AnyType:$input, NonConstClockType:$clock);
+  let results = (outs);
+
+  let assemblyFormat = "$clock `,` $input attr-dict `:` type($input)";
+}
```

Comment:
Grumpy old man nit: this exceeds 80 columns.

Reply:
We should enable clang-format on tablegen across CIRCT

Reply:
I thought tablegen was not supported by clang-format. It would be great if this worked!